### PR TITLE
Adds createWithDefault API

### LINKED ISSUE #34: Default value
That's be dope to get a default value, kind of a `startWith` that would be executed only once in the lifetime of the `replayingShare()`.

Right now, say I have

```kotlin
api.doStuff()
  .startWith(defaultData)
  .replayingShare()
```

the `defaultData` will be emitted for all new first subscriber.

### PR DESCRIPTION
**Pull request by oldergod**
fixes #34

### CODE REVIEW

File: replaying-share-kotlin/src/main/kotlin/com/jakewharton/rx/ReplayingShare.kt [hunk 1]
```diff
@@ -25,8 +25,12 @@ import io.reactivex.Observable
  * value from the upstream observable *only* when one or more downstream observers are connected.
  * This allows expensive upstream observables to be shut down when no one is observing while also
  * replaying the last value seen by *any* observer to new ones.
+ *
+ * @param defaultValue the initial value to be cached
  */
-fun <T> Observable<T>.replayingShare(): Observable<T> = compose(ReplayingShare.instance<T>())
+fun <T> Observable<T>.replayingShare(defaultValue: T? = null): Observable<T> {
+  return compose(ReplayingShare.instance<T>())
```
Comment (JakeWharton) [inline replaying-share-kotlin/src/main/kotlin/com/jakewharton/rx/ReplayingShare.kt]: Needs to be like
```kotlin
compose(if (defaultValue != null) ReplayingShare.createWithDefault(defaultValue) else ReplayingShare.instance())
```
and below

File: replaying-share/src/main/java/com/jakewharton/rx/ReplayingShare.java [hunk 2]
```diff
@@ -35,6 +36,9 @@
  */
 public final class ReplayingShare<T>
     implements ObservableTransformer<T, T>, FlowableTransformer<T, T> {
+
+  private final AtomicReference<T> value;
```
Comment (JakeWharton) [inline replaying-share/src/main/java/com/jakewharton/rx/ReplayingShare.java]: This never changes its value so it doesn't need to be atomic. Also please move down below static methods.

File: replaying-share/src/main/java/com/jakewharton/rx/ReplayingShare.java [hunk 3]
```diff
@@ -43,32 +47,58 @@
...
 
+  /**
+   * Creates a `ReplayingShare` transformer with a default value which will be emitted downstream
+   * on subscription if there is not any cached value yet.
+   */
+  public static <T> ReplayingShare<T> createWithDefault(T defaultValue) {
+    return new ReplayingShare<>(defaultValue);
+  }
+
   private ReplayingShare() {
```
Comment (JakeWharton) [inline replaying-share/src/main/java/com/jakewharton/rx/ReplayingShare.java]: Let's delete this constructor and move the non-null check into `createWithDefault`.

File: replaying-share-kotlin/src/main/kotlin/com/jakewharton/rx/ReplayingShare.kt [hunk 4]
```diff
@@ -25,8 +25,12 @@ import io.reactivex.Observable
  * value from the upstream observable *only* when one or more downstream observers are connected.
  * This allows expensive upstream observables to be shut down when no one is observing while also
  * replaying the last value seen by *any* observer to new ones.
+ *
+ * @param defaultValue the initial value to be cached
  */
-fun <T> Observable<T>.replayingShare(): Observable<T> = compose(ReplayingShare.instance<T>())
+fun <T> Observable<T>.replayingShare(defaultValue: T? = null): Observable<T> {
```
Comment (JakeWharton) [inline replaying-share-kotlin/src/main/kotlin/com/jakewharton/rx/ReplayingShare.kt]: Need to add `@JvmOverloads` to both to maintain binary compatibility, unfortunately.

**Comment by JakeWharton** [comment]
Rebase to pick up CI fix

**Comment by oldergod** [comment]
Renamed `value` to `defaultValue` and applied the rest.

File: replaying-share/src/main/java/com/jakewharton/rx/ReplayingShare.java [hunk 5]
```diff
@@ -35,40 +35,64 @@
...
     return new LastSeenFlowable<>(upstream.doOnEach(lastSeen).share(), lastSeen);
   }
 
   static final class LastSeen<T> implements Observer<T>, Subscriber<T> {
+    private final T defaultValue;
     volatile T value;
 
+    LastSeen(T defaultValue) {
+      this.defaultValue = defaultValue;
+      value = this.defaultValue;
```
Comment (JakeWharton) [inline replaying-share/src/main/java/com/jakewharton/rx/ReplayingShare.java]: super nit:
```suggestion
      value =defaultValue;
```

Comment (JakeWharton) [reply replaying-share/src/main/java/com/jakewharton/rx/ReplayingShare.java]: GitHub won't let me apply this myself... stuipd

**Comment by JakeWharton** [comment]
Going to do a pass of nullability annotations and maybe add error-prone/nullaway before release.

**Comment by oldergod** [comment]
No problem, I can wait a few hours

**Pull request merged**